compiler/ballerina-lang/src/main/resources/grammar/BallerinaParser.g4

```diff
@@ -918,15 +923,15 @@ doClause
     ;
 
 queryPipeline
-    :   fromClause (fromClause | letClause | whereClause)*
+    :   fromClause ((fromClause | letClause | whereClause )* | (joinClause onClause)?)
```

Comment:
Is onClause mandatory for a join?

Reply:
isn't it?
https://github.com/ballerina-platform/ballerina-lang/issues/22382
it contains the join condition